web/server/watch/integration_test.go

```diff
@@ -121,15 +122,17 @@ func TestWatcher(t *testing.T) {
 				input <- stop
 			}()
 
-			results := []messaging.Folders{}
+			var results []messaging.Folders
 			for result := range output {
 				results = append(results, result)
 			}
 
-			So(len(results), ShouldEqual, 3)
-			So(len(results[0]), ShouldEqual, 2) // initial
-			So(len(results[1]), ShouldEqual, 1) // root moved to sub
-			So(len(results[2]), ShouldEqual, 2) // root moved back to original location
+			So(results, ShouldHaveLength, 5)
+			So(results[0], ShouldHaveLength, 2) // initial
+			So(results[1], ShouldHaveLength, 1) // root moved to sub
+			So(results[2], ShouldHaveLength, 1) // additional scan happens
+			So(results[3], ShouldHaveLength, 2) // root moved back to original location
+			So(results[3], ShouldHaveLength, 2) // additional scan happens
```

Comment:
this change is inconsequential, and the test is now deterministic

Reply:
hm... I'm not seeing any action happening between these two asserts to cause an additional scan though?

Reply:
I was just meaning that, specifically, these two testing lines have "zero" time between them - I would delete one of them, unless I'm missing something? Or maybe it should be `results[4]`?

Reply:
oh, oops, you're totally right. That should've been `results[4]`. Fixed